spec.html

```diff
@@ -19616,6 +19642,19 @@ <h1>Static Semantics: ContainsUseStrict</h1>
       </emu-alg>
     </emu-clause>
 
+    <emu-clause id="sec-function-definitions-static-semantics-hassourcetextavailable">
+      <h1>Static Semantics: HasSourceTextAvailable</h1>
+      <emu-see-also-para op="HasSourceTextAvailable"></emu-see-also-para>
+      <emu-grammar>FunctionBody : FunctionStatementList</emu-grammar>
+      <emu-alg>
+        1. If the Directive Prologue of |FunctionBody| contains a Sensitive Directive or a Hidden Implementation Directive, return *false*.
+        1. If |FunctionBody| occurs within a |ScriptBody|, |ModuleBody|, or |FunctionBody| that has a Directive Prologue that contains a Sensitive Directive or a Hidden Implementation Directive, return *false*.
```

Comment:
```suggestion
        1. If the Directive Prologue of |FunctionBody| contains a Hidden Implementation Directive, return *false*.
        1. If |FunctionBody| occurs within a |ScriptBody|, |ModuleBody|, or |FunctionBody| that has a Directive Prologue that contains a Hidden Implementation Directive, return *false*.
```
also moved to the second commit